src/main/java/io/phasetwo/service/model/jpa/JpaOrganizationProvider.java

```diff
@@ -47,9 +47,20 @@ public JpaOrganizationProvider(KeycloakSession session, EntityManager em) {
   @Override
   public OrganizationModel createOrganization(
       RealmModel realm, String name, UserModel createdBy, boolean admin) {
+    return createOrganization(realm, KeycloakModelUtils.generateId(), name, createdBy, admin);
+  }
+
+  @Override
+  public OrganizationModel createOrganization(
+      RealmModel realm, String id, String name, UserModel createdBy, boolean admin) {
     ExtOrganizationEntity e = new ExtOrganizationEntity();
-    e.setId(KeycloakModelUtils.generateId());
+    if (id == null) {
+      id = KeycloakModelUtils.generateId();
+    }
+    e.setId(id);
```

Comment:
This should throw an `IllegalArgumentException` if the value isn't specified. The above `createOrganization` method takes care of creating an `id` when it's not specified. If the caller is using this method, they should know they used it wrong.